app/src/main/java/org/schabi/newpipe/settings/SettingsResourceRegistry.java

```diff
@@ -0,0 +1,151 @@
+package org.schabi.newpipe.settings;
+
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+import androidx.annotation.XmlRes;
+import androidx.fragment.app.Fragment;
+
+import org.schabi.newpipe.R;
+
+import java.util.HashSet;
+import java.util.Objects;
+import java.util.Set;
+
+/**
+ * A registry that contains information about SettingsFragments.
+ * <br/>
+ * includes:
+ * <ul>
+ *     <li>Class of the SettingsFragment</li>
+ *     <li>XML-Resource</li>
+ *     <li>...</li>
+ * </ul>
+ *
+ * E.g. used by the preference search.
+ */
+public final class SettingsResourceRegistry {
+
+    private static final SettingsResourceRegistry INSTANCE = new SettingsResourceRegistry();
+
+    private final Set<SettingRegistryEntry> registeredEntries = new HashSet<>();
+
+    private SettingsResourceRegistry() {
+        add(MainSettingsFragment.class, R.xml.main_settings).setSearchable(false);
+
+        add(AppearanceSettingsFragment.class, R.xml.appearance_settings);
+        add(ContentSettingsFragment.class, R.xml.content_settings);
+        add(DownloadSettingsFragment.class, R.xml.download_settings);
+        add(HistorySettingsFragment.class, R.xml.history_settings);
+        add(NotificationSettingsFragment.class, R.xml.notification_settings);
+        add(UpdateSettingsFragment.class, R.xml.update_settings);
+        add(VideoAudioSettingsFragment.class, R.xml.video_audio_settings);
+    }
```

Comment:
What about the debug settings? I'd add them if `DEBUG == true`

Reply:
Done